src/server/api/routers/results.ts

```diff
@@ -249,96 +259,143 @@ async function generateImpactPayouts(round: Round, db: PrismaClient) {
+    for (const projectId in awardPayouts) {
+      if (awardPayouts[projectId]) {
+        awardPayouts[projectId].allocations =
+          (awardPayouts[projectId].allocations * amount) /
+          totalMetricsAmount;
 
-    const uniqueVoters = new Set(projectAllocations.map(a => a.voterId));
-    const projectVoters = uniqueVoters.size;
-    const projectId = projectMetric.project_id;
+        awardPayouts[projectId].allocationPercentage =
+          (awardPayouts[projectId].allocations * 100) / amount;
+      }
+    }
 
-    // Store the results in BallotResults format
-    projectPayouts[projectId] = {
-      voters: projectVoters,
-      allocations: projectTotalPayout, // This needs to be normalized
-    };
+    // Store the award payouts
+    projectPayouts[award.id] = awardPayouts;
   }
 
-  console.log("projectPayouts before normalization", projectPayouts);
+  console.log("projectPayouts by award (after normalized)", projectPayouts);
+
+  // Combine payouts across all awards for each project
+  const combinedPayouts: BallotResults = {};
+
+  for (const awardId in projectPayouts) {
+    const awardPayouts = projectPayouts[awardId];
+    for (const projectId in awardPayouts) {
+      if (!awardPayouts[projectId]) continue;
+
+      if (!combinedPayouts[projectId]) {
+        combinedPayouts[projectId] = {
+          voters: awardPayouts[projectId].voters,
+          allocations: awardPayouts[projectId].allocations,
+        };
+      } else {
+        combinedPayouts[projectId].voters += awardPayouts[projectId].voters;
+        combinedPayouts[projectId].allocations +=
+          awardPayouts[projectId].allocations;
+      }
 
-  // Adjust each project's payout based on the scaling factor
-  for (const projectName in projectPayouts) {
-    if (projectPayouts[projectName]) {
-      projectPayouts[projectName].allocations = projectPayouts[projectName].allocations * 100 / totalMetricsAmount
+      combinedPayouts[projectId].allocationPercentage = (combinedPayouts[projectId].allocations * 100) / totalAmountForRound;
     }
   }
 
-  console.log("projectPayouts after normalization", projectPayouts);
+  console.log("totalAmountForRound", totalAmountForRound);
+  console.log("combinedPayouts", combinedPayouts);
 
   // Compute additional metrics
-  const normalizedTotalVotes = Object.values(projectPayouts).reduce(
+  const normalizedTotalVotes = Object.values(combinedPayouts).reduce(
     (sum, result) => sum + result.allocations,
-    0
+    0,
   );
-  const totalVoters = Object.values(projectPayouts).reduce(
+  console.log("normalizedTotalVotes", normalizedTotalVotes);
+  
+  const totalVoters = Object.values(combinedPayouts).reduce(
     (sum, result) => sum + result.voters,
-    0
+    0,
   );
+  console.log("totalVoters", totalVoters);
+  
   const averageVotes = totalVoters > 0 ? normalizedTotalVotes / totalVoters : 0;
+  console.log("averageVotes", averageVotes);
 
   return {
-    votes: projectPayouts,
+    votes: combinedPayouts,
```

Comment:
we should also consider to return the awardPayouts -> i think the leaderboard will show the the leading projects by awards

Reply:
True that was set up for a follow up PR